qrest/resource.py

```diff
@@ -597,14 +602,24 @@ def __init__(
             to contain the previously obtained subsection of the json tree
         """
 
+        self.initial_kwargs = {
+            "extract_section": extract_section,
+            "create_attribute": create_attribute,
+        }
         self.response = JSONResponse(extract_section, create_attribute)
 
+    def clone(self):
+        """Return a new instance created with the same parameters as the current one."""
+        return self.__class__(**self.initial_kwargs)
 
-class CSVResource(Resource):
-    """ A REST Resource that expects a text/csv return
 
-    """
+class CSVResource(Resource):
+    """A REST Resource that expects a text/csv return"""
 
     def __init__(self):
         """Set the use of a CSVResponse."""
         self.response = CSVResponse()
+
+    def clone(self):
```

Comment:
considering this functionality is basic, why not make this default instead of forcing an abstractmethod?

Reply:
I hope I understand your comment correctly: you want a default clone method in the Response parent class. 

This also crossed my mind. The problem is that there isn't a default way to construct an instance of a Response subclass: most of them are created without parameters (to their init dunder), but not all. Note that EDI also provides several subclasses. 

By making it an abstract method, the writer of the Response subclass becomes acutely aware of the fact that they need to create a  clone method that passes the right parameters to create a new instance. There is nothing to automatically fallback to and that might hide an error.